WordPress/Classes/Utility/BackgroundTasks/WeeklyRoundupBackgroundTask.swift

```diff
@@ -234,11 +234,17 @@ class WeeklyRoundupBackgroundTask: BackgroundTask {
     /// Just a convenience method to know then this task is run, what run date to use as "current".
     ///
     private func currentRunPeriodEndDate() -> Date {
-        Calendar.current.nextDate(
+        let runDate = Calendar.current.nextDate(
             after: Date(),
             matching: runDateComponents,
             matchingPolicy: .nextTime,
             direction: .backward) ?? Date()
+
+        // The run date is when the task is scheduled to run, but the period end date is actually
+        // the previous day at 24:59:59.
+        let periodEndDate = Calendar.current.date(bySettingHour: 0, minute: 0, second: 0, of: runDate)!.addingTimeInterval(TimeInterval.init(-1))
```

Comment:
Left [a comment about that change in the original PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/17144#pullrequestreview-753681377) as this change piked my curiosity (as I'm particularly wary and cautious about some calendar fallacies that we all often fall into 😛 ).

(That PR introducing this is already merged but I figured it was a better place to ask the question to Diego anyway than asking here, as this is not what this release->develop PR is about.)